### PR TITLE
Add Redis Client, Initial cut at session implementation

### PR DESCRIPTION
**Pull request by golanglemonade**
- Adds a redis client (based on `go-redis`), first use case being hold user sessions
- Adds redis to compose setups
- Adds redis connection to ready checks:
```
{
  "fga": "OK",
  "redis": "OK",
  "sqlite_db_primary": "OK"
}
```
- Adds a sessions package to set + save user sessions in cookies + redis  with current expiration of 10 minutes
- Uses the CookieManager cookies for the setting of session cookies
- Saves sessions when using the CLI in the user keychain so subsequent requests will have the session cookie
- Sessions returned in the cookie are encrypted using `gorilla/securecookie`
- Sessions stored in redis will be the decrypted value with the `key` as the `user.ID`
```
127.0.0.1:6379> get 01HMDBSNA69X57KAKXM0TV1DZZ
"519ad487-9e17-46f4-a478-073d4c61dce8"
```
- Adds a debug logger interceptor, useful for printing out request headers and boy, e.g :
```
go run cmd/cli/main.go org get -i 01HMDBSNBFH0YCACHJ4FM24YE1  --debug
Request header sent: map[Accept:[application/json; charset=utf-8] Authorization:[Bearer eyJhbGciOiJSUzI1NiIsImtpZCI6IjAxSEhBUzY3QU03Mzc3OFMwUUVaM0NFQUdFIiwidHlwIjoiSldUIn0.eyJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjE3NjA4Iiwic3ViIjoiMDFITURCU05BNjlYNTdLQUtYTTBUVjFEWloiLCJhdWQiOlsiaHR0cDovL2xvY2FsaG9zdDoxNzYwOCJdLCJleHAiOjE3MDU3MTYwNzYsIm5iZiI6MTcwNTcxMjQ3NiwiaWF0IjoxNzA1NzEyNDc2LCJqdGkiOiIwMWhtajY1cnIzdm43ZnJlZTkyamJ5a3k1aCIsInVzZXJfaWQiOiIwMUhNREJTTkE2OVg1N0tBS1hNMFRWMURaWiIsImVtYWlsIjoic2Z1bmsrdGVzdEBkYXR1bS5uZXQiLCJvcmciOiIwMUhNREJTTkJGSDBZQ0FDSEo0Rk0yNFlFMSIsImRpc3BsYXlOYW1lIjoic2Z1bmsrdGVzdEBkYXR1bS5uZXQiLCJhdmF0YXJVUkwiOiJodHRwczovL3d3dy5ncmF2YXRhci5jb20vYXZhdGFyLzVlNjViMGVlMTFhN2UxYmI0ZmI3MTI0YWM0YmU1NTVmP2Q9cm9ib2hhc2hcdTAwMjZyPXBnXHUwMDI2cz04MCJ9.AuZwpRKez9capbCzK8xFYPnel9oYPSIFV03NsFKnIpgJxCs-LOaLGCB7VvzsyUXqBg8sr4hj4d2mh3vmK4jy64AK652KPAcNT6N-aozOPl1_0cgjdzUExgXmnXZPkGJlQyWktphkIge-OAwegrx7Yr8TUplpDQzsYy0PYiUOfwPIAA_7OrQjTMJca_2Yh1GStfRLz9N0v_kzPrSYI3AH_rUj2AeAqVSxzWdZOYxwqGPHdDNlEkkS4T4I-eRci0KZkHgYGuOQ8xdMNAM4BdDLJGjH82rlQpMZK8LVKNpLB4F1qznCloke0n_Jm9SHUSTA9xuALBeAAhyIpI9vsFcV0A] Content-Type:[application/json; charset=utf-8] Cookie:[__session_id=MTcwNTcxMjQ4MHxCcTNOLVZreWpCQXN0anQtaVR3ZExPY0FYUnA3WU9MenM4MVFsamZVQzBvNzhtZnFpdF9nZERkanNzVjlfMlB2TGpZaEhfRnYyWm9STkVJWlFtVTJfZmJCYW5rdDdva09zSWZ3RzB1TEpfZFhuQkxHSTlwMG1KWW9zRE5RYTFzT2lHZjR8S51X3jufmWzNYX1Wp3kcGOfGkNXPp0i2Zd5tQeZ4w_o=]]
Request body sent: {{"query":"query GetOrganizationByID ($organizationId: ID!) {\n\torganization(id: $organizationId) {\n\t\tid\n\t\tname\n\t\tdisplayName\n\t\tdescription\n\t\tpersonalOrg\n\t\tparent {\n\t\t\tid\n\t\t\tname\n\t\t}\n\t\tchildren {\n\t\t\tedges {\n\t\t\t\tnode {\n\t\t\t\t\tid\n\t\t\t\t\tname\n\t\t\t\t\tdisplayName\n\t\t\t\t\tdescription\n\t\t\t\t}\n\t\t\t}\n\t\t}\n\t\tmembers {\n\t\t\tid\n\t\t\trole\n\t\t\tuser {\n\t\t\t\tid\n\t\t\t\tfirstName\n\t\t\t\tlastName\n\t\t\t}\n\t\t}\n\t\tsetting {\n\t\t\tid\n\t\t\tcreatedAt\n\t\t\tupdatedAt\n\t\t\tcreatedBy\n\t\t\tupdatedBy\n\t\t\tdomains\n\t\t\tssoCert\n\t\t\tssoEntrypoint\n\t\t\tssoIssuer\n\t\t\tbillingContact\n\t\t\tbillingEmail\n\t\t\tbillingPhone\n\t\t\tbillingAddress\n\t\t\ttaxIdentifier\n\t\t\ttags\n\t\t}\n\t\tcreatedAt\n\t\tcreatedBy\n\t\tupdatedAt\n\t\tupdatedBy\n\t}\n}\n","variables":{"organizationId":"01HMDBSNBFH0YCACHJ4FM24YE1"},"operationName":"GetOrganizationByID"}}
```

**NOTE**: This breaks all `--no-auth` calls, but this functionality will be removed in a follow-up PR because its not worth the maintance when the majority of the functionality requires information about a user

### CODE REVIEW

File: internal/sessions/sessions.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
+package sessions
+
+import (
+	"context"
+	"net/http"
+	"sync"
+
+	"github.com/google/uuid"
```
Comment (matoszz) [inline internal/sessions/sessions.go:8]: just a nit, but would prefer to use the ulids package like we use for all our other ID's

File: config/.env-devauth.example [hunk 2]
```diff
@@ -25,6 +25,10 @@ DATUM_AUTHZ_HOST=localhost:8080
 DATUM_AUTHZ_SCHEME=http
 DATUM_AUTHZ_CREATE_NEW_MODEL=false
 
+# Session Settings
+DATUM_SESSIONS_SIGNING_KEY=4N0txRe4PSBTIGhXpMEriFlfDEH9zTri
```
Comment (matoszz) [inline config/.env-devauth.example:29]: how were these generated? are there constraints on length, etc.?

**Review by matoszz — APPROVED** [review]
just a couple of nits, otherwise this looks great!

File: config/.env-devauth.example (hunk 2 above)
Comment (golanglemonade) [reply config/.env-devauth.example:29]: Just a random string, it has to be 16, 32, or 64 characters long. I'll make sure thats documented before merging this.

File: internal/sessions/sessions.go (hunk 1 above)
Comment (golanglemonade) [reply internal/sessions/sessions.go:8]: good call, I'll update

Comment (golanglemonade) [reply internal/sessions/sessions.go]:
```
127.0.0.1:6379> get 01HMMH5AF96G3E7CXQTFT86NG9
"01HMMVHDYDHNQFRRMP9CF86147"
```

**Review by matoszz — APPROVED** [review]

**Pull request merged**